app/javascript/controllers/teams_tab_change_controller.js

```diff
@@ -1,34 +1,23 @@
 import { Controller } from "@hotwired/stimulus";
 
 export default class extends Controller {
-  static targets = ["teamContent", "allUsersSection", "showAllLink"];
+  static targets = ["teamContent"];
+
   changeTab(event) {
     event.preventDefault();
     const selectedTab = event.currentTarget.dataset.tab;
-    this.showTab(selectedTab);
-    this.setActiveTab(selectedTab);
-  }
 
-  showTab(tab) {
     this.teamContentTargets.forEach((el) => {
-      el.classList.add("hidden");
+      el.classList.toggle("hidden", el.dataset.tab !== selectedTab);
     });
-    const selectedContent = this.teamContentTargets.find(target => target.dataset.tab === tab);
-    if (selectedContent) {
-      selectedContent.classList.remove("hidden");
-    }
+
+    this.setActiveTab(selectedTab);
   }
+
   setActiveTab(activeTab) {
     const tabs = this.element.querySelectorAll("a[data-tab]");
```

Comment:
Can we avoid querySelector here and rely only on targets

Reply:
Query selector Removed and   used " const selectedTab = event.currentTarget.dataset.tab;"